lib/rex/post/meterpreter/client_core.rb

```diff
@@ -768,9 +768,16 @@ def negotiate_tlv_encryption(timeout: client.comm_timeout)
       response = client.send_request(request, timeout)
       key_enc = response.get_tlv_value(TLV_TYPE_ENC_SYM_KEY)
       key_type = response.get_tlv_value(TLV_TYPE_SYM_KEY_TYPE)
-
+      is_weak_key = false
       if key_enc
-        sym_key = rsa_key.private_decrypt(key_enc, OpenSSL::PKey::RSA::PKCS1_PADDING)
+        key_dec_data = rsa_key.private_decrypt(key_enc, OpenSSL::PKey::RSA::PKCS1_PADDING)
+
+        if key_dec_data.length == 17 || key_dec_data.length == 33
+          sym_key = key_dec_data[0, key_dec_data.length - 1]
+          is_weak_key = key_dec_data[key_dec_data.length - 1] != "\x00"
+        else
+          sym_key = key_dec_data
+        end
```

Comment:
I think we might want to make this a bit more explicit to avoid any unintended behavior

Something like
```
key_length = { AES128 => 16, AES256 => 32 }[key_type]
# fail somehow if key_length.nil? (bad key_type)

sym_key = key_dec_data[...key_length]
key_dec_data = key_dec_data[key_length...]
# fail somehow if sym_key.length != key_length

if key_dec_data.length > 0
  key_strength = key_dec_data[0]
  key_dec_data = key_dec_data[1...]
  is_weak_key = key_strength != "\x00"
end
```
By the end, `key_dec_data` should be fully consumed and be empty, if not that means extra data has been added so let's ignore it because it's probably some future feild that was also appended. As it is, if we add another field in the future, the == 17 || == 33 logic wouldn't catch it because we'd be looking at a length of 18 or 34 meaning the session wouldn't establish.